// @angular/platform-browser/animations/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


export {ɵWebAnimationsStyleNormalizer as ɵWebAnimationsStyleNormalizer_7,AnimationDriver as AnimationDriver_10,ɵAnimationStyleNormalizer as ɵAnimationStyleNormalizer_11,ɵAnimationEngine as ɵAnimationEngine_12,ɵNoopAnimationEngine as ɵNoopAnimationEngine_17} from '@angular/animations/browser';
import * as import0 from '@angular/core';
import * as import1 from '@angular/platform-browser/animations';
import * as import2 from '@angular/common';
import * as import3 from '@angular/platform-browser';
import * as import4 from '@angular/animations/browser';
class BrowserAnimationsModuleInjector extends import0.ɵNgModuleInjector<import1.BrowserAnimationsModule> {
  _CommonModule_0:import2.CommonModule;
  _ErrorHandler_1:any;
  _APP_INITIALIZER_2:any[];
  _ApplicationInitStatus_3:import0.ApplicationInitStatus;
  _ɵf_4:import0.ɵf;
  _ApplicationRef_5:any;
  _ApplicationModule_6:import0.ApplicationModule;
  _BrowserModule_7:import3.BrowserModule;
  _BrowserAnimationsModule_8:import1.BrowserAnimationsModule;
  __LOCALE_ID_9:any;
  __NgLocalization_10:import2.NgLocaleLocalization;
  __Compiler_11:import0.Compiler;
  __APP_ID_12:any;
  __IterableDiffers_13:any;
  __KeyValueDiffers_14:any;
  __DomSanitizer_15:import3.ɵe;
  __Sanitizer_16:any;
  __HAMMER_GESTURE_CONFIG_17:import3.HammerGestureConfig;
  __EVENT_MANAGER_PLUGINS_18:any[];
  __EventManager_19:import3.EventManager;
  __ɵDomSharedStylesHost_20:import3.ɵDomSharedStylesHost;
  __ɵDomRendererFactory2_21:import3.ɵDomRendererFactory2;
  __AnimationDriver_22:any;
  __ɵAnimationStyleNormalizer_23:any;
  __ɵAnimationEngine_24:import1.ɵa;
  __RendererFactory2_25:any;
  __ɵSharedStylesHost_26:any;
  __Testability_27:import0.Testability;
  __Meta_28:import3.Meta;
  __Title_29:import3.Title;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _LOCALE_ID_9():any {
    if ((this.__LOCALE_ID_9 == null)) { (this.__LOCALE_ID_9 = import0.ɵn(this.parent.get(import0.LOCALE_ID,(null as any)))); }
    return this.__LOCALE_ID_9;
  }
  get _NgLocalization_10():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_10 == null)) { (this.__NgLocalization_10 = new import2.NgLocaleLocalization(this._LOCALE_ID_9)); }
    return this.__NgLocalization_10;
  }
  get _Compiler_11():import0.Compiler {
    if ((this.__Compiler_11 == null)) { (this.__Compiler_11 = new import0.Compiler()); }
    return this.__Compiler_11;
  }
  get _APP_ID_12():any {
    if ((this.__APP_ID_12 == null)) { (this.__APP_ID_12 = import0.ɵg()); }
    return this.__APP_ID_12;
  }
  get _IterableDiffers_13():any {
    if ((this.__IterableDiffers_13 == null)) { (this.__IterableDiffers_13 = import0.ɵl()); }
    return this.__IterableDiffers_13;
  }
  get _KeyValueDiffers_14():any {
    if ((this.__KeyValueDiffers_14 == null)) { (this.__KeyValueDiffers_14 = import0.ɵm()); }
    return this.__KeyValueDiffers_14;
  }
  get _DomSanitizer_15():import3.ɵe {
    if ((this.__DomSanitizer_15 == null)) { (this.__DomSanitizer_15 = new import3.ɵe(this.parent.get(import3.DOCUMENT))); }
    return this.__DomSanitizer_15;
  }
  get _Sanitizer_16():any {
    if ((this.__Sanitizer_16 == null)) { (this.__Sanitizer_16 = this._DomSanitizer_15); }
    return this.__Sanitizer_16;
  }
  get _HAMMER_GESTURE_CONFIG_17():import3.HammerGestureConfig {
    if ((this.__HAMMER_GESTURE_CONFIG_17 == null)) { (this.__HAMMER_GESTURE_CONFIG_17 = new import3.HammerGestureConfig()); }
    return this.__HAMMER_GESTURE_CONFIG_17;
  }
  get _EVENT_MANAGER_PLUGINS_18():any[] {
    if ((this.__EVENT_MANAGER_PLUGINS_18 == null)) { (this.__EVENT_MANAGER_PLUGINS_18 = [
      new import3.ɵDomEventsPlugin(this.parent.get(import3.DOCUMENT)),
      new import3.ɵKeyEventsPlugin(this.parent.get(import3.DOCUMENT)),
      new import3.ɵHammerGesturesPlugin(this.parent.get(import3.DOCUMENT),this._HAMMER_GESTURE_CONFIG_17)
    ]
    ); }
    return this.__EVENT_MANAGER_PLUGINS_18;
  }
  get _EventManager_19():import3.EventManager {
    if ((this.__EventManager_19 == null)) { (this.__EventManager_19 = new import3.EventManager(this._EVENT_MANAGER_PLUGINS_18,this.parent.get(import0.NgZone))); }
    return this.__EventManager_19;
  }
  get _ɵDomSharedStylesHost_20():import3.ɵDomSharedStylesHost {
    if ((this.__ɵDomSharedStylesHost_20 == null)) { (this.__ɵDomSharedStylesHost_20 = new import3.ɵDomSharedStylesHost(this.parent.get(import3.DOCUMENT))); }
    return this.__ɵDomSharedStylesHost_20;
  }
  get _ɵDomRendererFactory2_21():import3.ɵDomRendererFactory2 {
    if ((this.__ɵDomRendererFactory2_21 == null)) { (this.__ɵDomRendererFactory2_21 = new import3.ɵDomRendererFactory2(this._EventManager_19,this._ɵDomSharedStylesHost_20)); }
    return this.__ɵDomRendererFactory2_21;
  }
  get _AnimationDriver_22():any {
    if ((this.__AnimationDriver_22 == null)) { (this.__AnimationDriver_22 = import1.ɵb()); }
    return this.__AnimationDriver_22;
  }
  get _ɵAnimationStyleNormalizer_23():any {
    if ((this.__ɵAnimationStyleNormalizer_23 == null)) { (this.__ɵAnimationStyleNormalizer_23 = import1.ɵc()); }
    return this.__ɵAnimationStyleNormalizer_23;
  }
  get _ɵAnimationEngine_24():import1.ɵa {
    if ((this.__ɵAnimationEngine_24 == null)) { (this.__ɵAnimationEngine_24 = new import1.ɵa(this._AnimationDriver_22,this._ɵAnimationStyleNormalizer_23)); }
    return this.__ɵAnimationEngine_24;
  }
  get _RendererFactory2_25():any {
    if ((this.__RendererFactory2_25 == null)) { (this.__RendererFactory2_25 = import1.ɵd(this._ɵDomRendererFactory2_21,this._ɵAnimationEngine_24,this.parent.get(import0.NgZone))); }
    return this.__RendererFactory2_25;
  }
  get _ɵSharedStylesHost_26():any {
    if ((this.__ɵSharedStylesHost_26 == null)) { (this.__ɵSharedStylesHost_26 = this._ɵDomSharedStylesHost_20); }
    return this.__ɵSharedStylesHost_26;
  }
  get _Testability_27():import0.Testability {
    if ((this.__Testability_27 == null)) { (this.__Testability_27 = new import0.Testability(this.parent.get(import0.NgZone))); }
    return this.__Testability_27;
  }
  get _Meta_28():import3.Meta {
    if ((this.__Meta_28 == null)) { (this.__Meta_28 = new import3.Meta(this.parent.get(import3.DOCUMENT))); }
    return this.__Meta_28;
  }
  get _Title_29():import3.Title {
    if ((this.__Title_29 == null)) { (this.__Title_29 = new import3.Title(this.parent.get(import3.DOCUMENT))); }
    return this.__Title_29;
  }
  createInternal():import1.BrowserAnimationsModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ErrorHandler_1 = import3.ɵa();
    this._APP_INITIALIZER_2 = [
      import0.ɵo,
      import3.ɵc(this.parent.get(import3.NgProbeToken,(null as any)),this.parent.get(import0.NgProbeToken,(null as any)))
    ]
    ;
    this._ApplicationInitStatus_3 = new import0.ApplicationInitStatus(this._APP_INITIALIZER_2);
    this._ɵf_4 = new import0.ɵf(this.parent.get(import0.NgZone),this.parent.get(import0.ɵConsole),this,this._ErrorHandler_1,this.componentFactoryResolver,this._ApplicationInitStatus_3);
    this._ApplicationRef_5 = this._ɵf_4;
    this._ApplicationModule_6 = new import0.ApplicationModule(this._ApplicationRef_5);
    this._BrowserModule_7 = new import3.BrowserModule(this.parent.get(import3.BrowserModule,(null as any)));
    this._BrowserAnimationsModule_8 = new import1.BrowserAnimationsModule();
    return this._BrowserAnimationsModule_8;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import0.ErrorHandler)) { return this._ErrorHandler_1; }
    if ((token === import0.APP_INITIALIZER)) { return this._APP_INITIALIZER_2; }
    if ((token === import0.ApplicationInitStatus)) { return this._ApplicationInitStatus_3; }
    if ((token === import0.ɵf)) { return this._ɵf_4; }
    if ((token === import0.ApplicationRef)) { return this._ApplicationRef_5; }
    if ((token === import0.ApplicationModule)) { return this._ApplicationModule_6; }
    if ((token === import3.BrowserModule)) { return this._BrowserModule_7; }
    if ((token === import1.BrowserAnimationsModule)) { return this._BrowserAnimationsModule_8; }
    if ((token === import0.LOCALE_ID)) { return this._LOCALE_ID_9; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_10; }
    if ((token === import0.Compiler)) { return this._Compiler_11; }
    if ((token === import0.APP_ID)) { return this._APP_ID_12; }
    if ((token === import0.IterableDiffers)) { return this._IterableDiffers_13; }
    if ((token === import0.KeyValueDiffers)) { return this._KeyValueDiffers_14; }
    if ((token === import3.DomSanitizer)) { return this._DomSanitizer_15; }
    if ((token === import0.Sanitizer)) { return this._Sanitizer_16; }
    if ((token === import3.HAMMER_GESTURE_CONFIG)) { return this._HAMMER_GESTURE_CONFIG_17; }
    if ((token === import3.EVENT_MANAGER_PLUGINS)) { return this._EVENT_MANAGER_PLUGINS_18; }
    if ((token === import3.EventManager)) { return this._EventManager_19; }
    if ((token === import3.ɵDomSharedStylesHost)) { return this._ɵDomSharedStylesHost_20; }
    if ((token === import3.ɵDomRendererFactory2)) { return this._ɵDomRendererFactory2_21; }
    if ((token === import4.AnimationDriver)) { return this._AnimationDriver_22; }
    if ((token === import4.ɵAnimationStyleNormalizer)) { return this._ɵAnimationStyleNormalizer_23; }
    if ((token === import4.ɵAnimationEngine)) { return this._ɵAnimationEngine_24; }
    if ((token === import0.RendererFactory2)) { return this._RendererFactory2_25; }
    if ((token === import3.ɵSharedStylesHost)) { return this._ɵSharedStylesHost_26; }
    if ((token === import0.Testability)) { return this._Testability_27; }
    if ((token === import3.Meta)) { return this._Meta_28; }
    if ((token === import3.Title)) { return this._Title_29; }
    return notFoundResult;
  }
  destroyInternal():void {
    this._ɵf_4.ngOnDestroy();
    (this.__ɵDomSharedStylesHost_20 && this._ɵDomSharedStylesHost_20.ngOnDestroy());
  }
}
export const BrowserAnimationsModuleNgFactory:import0.NgModuleFactory<import1.BrowserAnimationsModule> = new import0.NgModuleFactory<any>(BrowserAnimationsModuleInjector,import1.BrowserAnimationsModule);
class NoopAnimationsModuleInjector extends import0.ɵNgModuleInjector<import1.NoopAnimationsModule> {
  _CommonModule_0:import2.CommonModule;
  _ErrorHandler_1:any;
  _APP_INITIALIZER_2:any[];
  _ApplicationInitStatus_3:import0.ApplicationInitStatus;
  _ɵf_4:import0.ɵf;
  _ApplicationRef_5:any;
  _ApplicationModule_6:import0.ApplicationModule;
  _BrowserModule_7:import3.BrowserModule;
  _NoopAnimationsModule_8:import1.NoopAnimationsModule;
  __LOCALE_ID_9:any;
  __NgLocalization_10:import2.NgLocaleLocalization;
  __Compiler_11:import0.Compiler;
  __APP_ID_12:any;
  __IterableDiffers_13:any;
  __KeyValueDiffers_14:any;
  __DomSanitizer_15:import3.ɵe;
  __Sanitizer_16:any;
  __HAMMER_GESTURE_CONFIG_17:import3.HammerGestureConfig;
  __EVENT_MANAGER_PLUGINS_18:any[];
  __EventManager_19:import3.EventManager;
  __ɵDomSharedStylesHost_20:import3.ɵDomSharedStylesHost;
  __ɵDomRendererFactory2_21:import3.ɵDomRendererFactory2;
  __ɵAnimationEngine_22:import4.ɵNoopAnimationEngine;
  __RendererFactory2_23:any;
  __ɵSharedStylesHost_24:any;
  __Testability_25:import0.Testability;
  __Meta_26:import3.Meta;
  __Title_27:import3.Title;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _LOCALE_ID_9():any {
    if ((this.__LOCALE_ID_9 == null)) { (this.__LOCALE_ID_9 = import0.ɵn(this.parent.get(import0.LOCALE_ID,(null as any)))); }
    return this.__LOCALE_ID_9;
  }
  get _NgLocalization_10():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_10 == null)) { (this.__NgLocalization_10 = new import2.NgLocaleLocalization(this._LOCALE_ID_9)); }
    return this.__NgLocalization_10;
  }
  get _Compiler_11():import0.Compiler {
    if ((this.__Compiler_11 == null)) { (this.__Compiler_11 = new import0.Compiler()); }
    return this.__Compiler_11;
  }
  get _APP_ID_12():any {
    if ((this.__APP_ID_12 == null)) { (this.__APP_ID_12 = import0.ɵg()); }
    return this.__APP_ID_12;
  }
  get _IterableDiffers_13():any {
    if ((this.__IterableDiffers_13 == null)) { (this.__IterableDiffers_13 = import0.ɵl()); }
    return this.__IterableDiffers_13;
  }
  get _KeyValueDiffers_14():any {
    if ((this.__KeyValueDiffers_14 == null)) { (this.__KeyValueDiffers_14 = import0.ɵm()); }
    return this.__KeyValueDiffers_14;
  }
  get _DomSanitizer_15():import3.ɵe {
    if ((this.__DomSanitizer_15 == null)) { (this.__DomSanitizer_15 = new import3.ɵe(this.parent.get(import3.DOCUMENT))); }
    return this.__DomSanitizer_15;
  }
  get _Sanitizer_16():any {
    if ((this.__Sanitizer_16 == null)) { (this.__Sanitizer_16 = this._DomSanitizer_15); }
    return this.__Sanitizer_16;
  }
  get _HAMMER_GESTURE_CONFIG_17():import3.HammerGestureConfig {
    if ((this.__HAMMER_GESTURE_CONFIG_17 == null)) { (this.__HAMMER_GESTURE_CONFIG_17 = new import3.HammerGestureConfig()); }
    return this.__HAMMER_GESTURE_CONFIG_17;
  }
  get _EVENT_MANAGER_PLUGINS_18():any[] {
    if ((this.__EVENT_MANAGER_PLUGINS_18 == null)) { (this.__EVENT_MANAGER_PLUGINS_18 = [
      new import3.ɵDomEventsPlugin(this.parent.get(import3.DOCUMENT)),
      new import3.ɵKeyEventsPlugin(this.parent.get(import3.DOCUMENT)),
      new import3.ɵHammerGesturesPlugin(this.parent.get(import3.DOCUMENT),this._HAMMER_GESTURE_CONFIG_17)
    ]
    ); }
    return this.__EVENT_MANAGER_PLUGINS_18;
  }
  get _EventManager_19():import3.EventManager {
    if ((this.__EventManager_19 == null)) { (this.__EventManager_19 = new import3.EventManager(this._EVENT_MANAGER_PLUGINS_18,this.parent.get(import0.NgZone))); }
    return this.__EventManager_19;
  }
  get _ɵDomSharedStylesHost_20():import3.ɵDomSharedStylesHost {
    if ((this.__ɵDomSharedStylesHost_20 == null)) { (this.__ɵDomSharedStylesHost_20 = new import3.ɵDomSharedStylesHost(this.parent.get(import3.DOCUMENT))); }
    return this.__ɵDomSharedStylesHost_20;
  }
  get _ɵDomRendererFactory2_21():import3.ɵDomRendererFactory2 {
    if ((this.__ɵDomRendererFactory2_21 == null)) { (this.__ɵDomRendererFactory2_21 = new import3.ɵDomRendererFactory2(this._EventManager_19,this._ɵDomSharedStylesHost_20)); }
    return this.__ɵDomRendererFactory2_21;
  }
  get _ɵAnimationEngine_22():import4.ɵNoopAnimationEngine {
    if ((this.__ɵAnimationEngine_22 == null)) { (this.__ɵAnimationEngine_22 = new import4.ɵNoopAnimationEngine()); }
    return this.__ɵAnimationEngine_22;
  }
  get _RendererFactory2_23():any {
    if ((this.__RendererFactory2_23 == null)) { (this.__RendererFactory2_23 = import1.ɵd(this._ɵDomRendererFactory2_21,this._ɵAnimationEngine_22,this.parent.get(import0.NgZone))); }
    return this.__RendererFactory2_23;
  }
  get _ɵSharedStylesHost_24():any {
    if ((this.__ɵSharedStylesHost_24 == null)) { (this.__ɵSharedStylesHost_24 = this._ɵDomSharedStylesHost_20); }
    return this.__ɵSharedStylesHost_24;
  }
  get _Testability_25():import0.Testability {
    if ((this.__Testability_25 == null)) { (this.__Testability_25 = new import0.Testability(this.parent.get(import0.NgZone))); }
    return this.__Testability_25;
  }
  get _Meta_26():import3.Meta {
    if ((this.__Meta_26 == null)) { (this.__Meta_26 = new import3.Meta(this.parent.get(import3.DOCUMENT))); }
    return this.__Meta_26;
  }
  get _Title_27():import3.Title {
    if ((this.__Title_27 == null)) { (this.__Title_27 = new import3.Title(this.parent.get(import3.DOCUMENT))); }
    return this.__Title_27;
  }
  createInternal():import1.NoopAnimationsModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ErrorHandler_1 = import3.ɵa();
    this._APP_INITIALIZER_2 = [
      import0.ɵo,
      import3.ɵc(this.parent.get(import3.NgProbeToken,(null as any)),this.parent.get(import0.NgProbeToken,(null as any)))
    ]
    ;
    this._ApplicationInitStatus_3 = new import0.ApplicationInitStatus(this._APP_INITIALIZER_2);
    this._ɵf_4 = new import0.ɵf(this.parent.get(import0.NgZone),this.parent.get(import0.ɵConsole),this,this._ErrorHandler_1,this.componentFactoryResolver,this._ApplicationInitStatus_3);
    this._ApplicationRef_5 = this._ɵf_4;
    this._ApplicationModule_6 = new import0.ApplicationModule(this._ApplicationRef_5);
    this._BrowserModule_7 = new import3.BrowserModule(this.parent.get(import3.BrowserModule,(null as any)));
    this._NoopAnimationsModule_8 = new import1.NoopAnimationsModule();
    return this._NoopAnimationsModule_8;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import0.ErrorHandler)) { return this._ErrorHandler_1; }
    if ((token === import0.APP_INITIALIZER)) { return this._APP_INITIALIZER_2; }
    if ((token === import0.ApplicationInitStatus)) { return this._ApplicationInitStatus_3; }
    if ((token === import0.ɵf)) { return this._ɵf_4; }
    if ((token === import0.ApplicationRef)) { return this._ApplicationRef_5; }
    if ((token === import0.ApplicationModule)) { return this._ApplicationModule_6; }
    if ((token === import3.BrowserModule)) { return this._BrowserModule_7; }
    if ((token === import1.NoopAnimationsModule)) { return this._NoopAnimationsModule_8; }
    if ((token === import0.LOCALE_ID)) { return this._LOCALE_ID_9; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_10; }
    if ((token === import0.Compiler)) { return this._Compiler_11; }
    if ((token === import0.APP_ID)) { return this._APP_ID_12; }
    if ((token === import0.IterableDiffers)) { return this._IterableDiffers_13; }
    if ((token === import0.KeyValueDiffers)) { return this._KeyValueDiffers_14; }
    if ((token === import3.DomSanitizer)) { return this._DomSanitizer_15; }
    if ((token === import0.Sanitizer)) { return this._Sanitizer_16; }
    if ((token === import3.HAMMER_GESTURE_CONFIG)) { return this._HAMMER_GESTURE_CONFIG_17; }
    if ((token === import3.EVENT_MANAGER_PLUGINS)) { return this._EVENT_MANAGER_PLUGINS_18; }
    if ((token === import3.EventManager)) { return this._EventManager_19; }
    if ((token === import3.ɵDomSharedStylesHost)) { return this._ɵDomSharedStylesHost_20; }
    if ((token === import3.ɵDomRendererFactory2)) { return this._ɵDomRendererFactory2_21; }
    if ((token === import4.ɵAnimationEngine)) { return this._ɵAnimationEngine_22; }
    if ((token === import0.RendererFactory2)) { return this._RendererFactory2_23; }
    if ((token === import3.ɵSharedStylesHost)) { return this._ɵSharedStylesHost_24; }
    if ((token === import0.Testability)) { return this._Testability_25; }
    if ((token === import3.Meta)) { return this._Meta_26; }
    if ((token === import3.Title)) { return this._Title_27; }
    return notFoundResult;
  }
  destroyInternal():void {
    this._ɵf_4.ngOnDestroy();
    (this.__ɵDomSharedStylesHost_20 && this._ɵDomSharedStylesHost_20.ngOnDestroy());
  }
}
export const NoopAnimationsModuleNgFactory:import0.NgModuleFactory<import1.NoopAnimationsModule> = new import0.NgModuleFactory<any>(NoopAnimationsModuleInjector,import1.NoopAnimationsModule);
